### PR TITLE
Feat/skribble UI

### PR DESCRIPTION
**Pull request by mokshgrappus007**

### CODE REVIEW

File: api/lib/session/bloc/session_bloc.dart [hunk 1]
```diff
@@ -49,7 +49,7 @@ class SessionBloc extends BroadcastBloc<SessionEvent, SessionState> {
   void _onMessageSent(OnMessageSent event, Emitter<SessionState> emit) {
     emit(
       state.copyWith(
-        messages: [...state.messages, event.chat],
+        messages: [event.chat,...state.messages],
```
Comment (noorbhatia) [inline api/lib/session/bloc/session_bloc.dart]: What's the reason for this change?

Comment (mokshgrappus007) [reply api/lib/session/bloc/session_bloc.dart]: did it coz in chat component need to show list in reverse order

File: packages/app_ui/pubspec.yaml [hunk 2]
```diff
@@ -17,3 +17,12 @@ dev_dependencies:
...
   uses-material-design: true
+  assets:
+    - assets/images/
+  fonts:
+    - family: OpenSans
+      fonts:
+        - asset: assets/fonts/OpenSans-Regular.ttf
+          weight: 400
+        - asset: assets/fonts/OpenSans-Bold.ttf
+          weight: 700
```
Comment (noorbhatia) [inline packages/app_ui/pubspec.yaml:28]: add medium(500) and semi bold(600) weights too, and remove unnecessary condensed font files

File: lib/views/game/view/leader_board.dart [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import 'package:grappus_skribbl/views/views.dart';
+
+class LeaderBoard extends StatelessWidget {
+  const LeaderBoard({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: 176.h,
+      width: MediaQuery.of(context).size.width,
```
Comment (noorbhatia) [inline lib/views/game/view/leader_board.dart]: extension for this can be added. And use MediaQuery.sizeof

File: pubspec.yaml [hunk 4]
```diff
@@ -25,6 +25,8 @@ dependencies:
     path: api/packages/player_repository
   app_ui:
     path: packages/app_ui
+  flutter_screenutil: ^5.9.0
```
Comment (noorbhatia) [inline pubspec.yaml]: It's better to not depend on third party packages for responsiveness and adaptiveness

File: api/lib/session/bloc/session_bloc.dart (hunk 1 above)
Comment (noorbhatia) [reply api/lib/session/bloc/session_bloc.dart]: We should not change the order of the chat on backend, it's the source of truth. Manipulate the chat on frontend

**Comment by noorbhatia** [comment]
@mokshgrappus007 resolve conflicts

File: lib/views/game/view/chat_component.dart [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+class ChatComponent extends StatefulWidget {
+  const ChatComponent({super.key});
+
+  @override
+  State<ChatComponent> createState() => _ChatComponentState();
+}
+
+class _ChatComponentState extends State<ChatComponent> {
+  final message = TextEditingController();
```
Comment (noorbhatia) [inline lib/views/game/view/chat_component.dart]: _chatController

File: api/packages/models/lib/web_socket_event.dart [hunk 6]
```diff
@@ -60,17 +60,14 @@ class AddToChatEvent extends WebSocketEvent<ChatModel> {
   }
 }
 
-class AddPlayerEvent extends WebSocketEvent<String> {
+class AddPlayerEvent extends WebSocketEvent<Map<String, String>> {
```
Comment (dhruvd-grappus) [inline api/packages/models/lib/web_socket_event.dart]: can we use `WebSocketEvent<Player>` instead?
Maps would be unsafe here.

File: api/routes/ws.dart [hunk 7]
```diff
@@ -48,7 +48,8 @@ Future<Response> onRequest(RequestContext context) async {
               sessionBloc.add(OnMessageSent(chatModel));
 
             case AddPlayerEvent:
-              player.name = (websocketEvent as AddPlayerEvent).data;
+              player.name = (websocketEvent as AddPlayerEvent).data['name']!;
```
Comment (dhruvd-grappus) [inline api/routes/ws.dart]: if `AddPlayerEvent` extends `WebSocketEvent<Player>`
then this would be simple:
`player=(websocketEvent as AddPlayerEvent).data;`

**Review by dhruvd-grappus — APPROVED** [review]
LGTM, squash commits before merging @mokshgrappus007
